client/network/src/behaviour.rs

```diff
@@ -172,6 +174,18 @@ pub enum BehaviourOut<B: BlockT> {
 	/// Events generated by a DHT as a response to get_value or put_value requests as well as the
 	/// request duration.
 	Dht(DhtEvent, Duration),
+
+	/// Event generated by bitswap.
+	Bitswap(BitswapEvent)
+}
+
+/// An event generated by bitswap.
+#[derive(Clone, Debug)]
+pub enum BitswapEvent {
+	/// A block was received.
+	ReceivedBlock(PeerId, tiny_cid::Cid, Box<[u8]>),
```

Comment:
Can this be called multiple times with the same block? What's the relationship between the `Cid` and the data? Is it guaranteed to be in response to a WANT request?

Reply:
yes

Reply:
> Can this be called multiple times with the same block?

no, unless the events weren't polled until the block arrived and a new want request was submitted.
> What's the relationship between the Cid and the data

the cid hash matches the data, the cid is the one submitted in the want request
>  Is it guaranteed to be in response to a WANT request?

it's guaranteed to be in response to a want request.